go/vt/vtgate/executor.go

```diff
@@ -1228,10 +1228,14 @@ func (e *Executor) getPlan(vcursor *vcursorImpl, sql string, comments sqlparser.
 		return nil, err
 	}
 
-	// Normalize if possible and retry.
 	query := sql
 	statement := stmt
 	bindVarNeeds := sqlparser.BindVarNeeds{}
+	if !isValidPayloadSize(query) && !sqlparser.MaxPayloadSizeOverrideDirective(statement) {
+		return nil, vterrors.New(vtrpcpb.Code_RESOURCE_EXHAUSTED, "query payload size above threshold")
```

Comment:
Should this be `Code_FAILED_PRECONDITION` instead?

Reply:
I think ResourceExhausted is fine, but it'd be nice to use a MySQL error code that indicates it's not retryable. In the upstream community Slack, we were discussing using MySQL error code 1153 for things like this and gRPC packet size exceeded going forward. I'm not sure where that MySQL error code is set and where the gRPC error code is set.